editor/editor_properties.cpp

```diff
@@ -2183,7 +2183,55 @@ void EditorPropertyResource::_menu_option(int p_which) {
 		case OBJ_MENU_NEW_SCRIPT: {
 
 			if (Object::cast_to<Node>(get_edited_object())) {
-				EditorNode::get_singleton()->get_scene_tree_dock()->open_script_dialog(Object::cast_to<Node>(get_edited_object()));
+				if (get_edited_property() == "script") {
+					EditorNode::get_singleton()->get_scene_tree_dock()->open_script_dialog(Object::cast_to<Node>(get_edited_object()));
+				} else {
+					// Copied from editor/scene_tree_dock.cpp#_tool_selected and edited for our purposes
+					// There's a good chance we could merge this back into that code with some modifications
+					Ref<Script> existing = get_edited_object()->get(get_edited_property());
+
+					String inherits;
+					if (existing.is_valid()) {
+						//inherits = existing->get_class();
+						for (int i = 0; i < ScriptServer::get_language_count(); i++) {
+							ScriptLanguage *l = ScriptServer::get_language(i);
+							if (l->get_type() == existing->get_class()) {
+								String name = l->get_global_class_name(existing->get_path());
+								if (ScriptServer::is_global_class(name) && EDITOR_GET("interface/editors/derive_script_globals_by_name").operator bool()) {
+									inherits = name;
+								} else if (l->can_inherit_from_file()) {
+									inherits = "\"" + existing->get_path() + "\"";
+								}
+								break;
+							}
+						}
+					} else {
+						inherits = "Reference";
+					}
+
+					Node *edited_node = Object::cast_to<Node>(get_edited_object());
+
+					String path = edited_node->get_filename().get_basename();
+					if (path == "") {
+						String root_path = EditorNode::get_singleton()->get_editor_data().get_edited_scene_root()->get_filename();
+						if (root_path == "") {
+							path = String("res://").plus_file(edited_node->get_name());
+						} else {
+							path = root_path.get_base_dir().plus_file(edited_node->get_name());
+						}
+					}
+					// path = <node_or_scene_name>.<property_name>.
+					// The extra period at the end is to prevent the property name from being seen as the type by ScriptCreateDialog
+					String path_property_name = get_edited_property();
+					path_property_name = path_property_name.replace(":", ".").replace("/", ".").replace("\\", ".");
+					path = path + "." + path_property_name + ".";
```

Comment:
As a bit of justification for why I did it this way: I can't just add the `.replace`s to the end of `get_edited_property()`, otherwise the compiler complains that I'm trying to use `.replace` on StringName, which doesn't have a `.replace` method despite being fully capable of being cast to a String.

Depending on how this gets merged into scene_tree_dock.cpp later, I may not need this weird set of three lines.